### PR TITLE
Shyrma bn mkl bp

### PR DESCRIPTION
**Pull request by shyrma**
provide batch normalization backprop operation based on corresponding mkl dnn api

### CODE REVIEW

File: libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp [hunk 1]
```diff
@@ -29,82 +29,6 @@ namespace nd4j {
...
-    Nd4jLong* outShapeInfo = nullptr;
-    const bool areShapesOk = ShapeUtils::evalCommonBroadcastShapeInfo(inArrs, outShapeInfo, block.getWorkspace());
-    REQUIRE_TRUE(areShapesOk, 0, "BATCHNORM op: the shapes of input arrays are not mutually broadcastable !");
-
-    auto result = ConstantShapeHelper::getInstance()->createShapeInfo(ShapeDescriptor(outShapeInfo, DataTypeUtils::pickFloatingType(ArrayOptions::dataType(in))));
-    return SHAPELIST(result);
-}
-
 //////////////////////////////////////////////////////////////////////////
 CUSTOM_OP_IMPL(batchnorm_new, 3, 1, false, 1, 2) {
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp]: since you had removed `batchnorm` - let's rename `batchnorm_new` to `batchnorm` now. 

p.s. and update mkldnn platform helper then. and java wrapper.

File: libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp [hunk 2]
```diff
@@ -176,298 +103,187 @@ DECLARE_SHAPE_FN(batchnorm_new) {
...
-        inArrs[i] = INPUT_VARIABLE(i);
-
-    // check whether all input shapes are mutually broadcastable
-    Nd4jLong* outShapeInfo = nullptr;
-    const bool areShapesOk = ShapeUtils::evalCommonBroadcastShapeInfo(inArrs, outShapeInfo, block.getWorkspace());
-    REQUIRE_TRUE(areShapesOk, 0, "BATCHNORM_BP op: the shapes of input arrays are not mutually broadcastable !");
+
+    std::string expShapeStr = ShapeUtils::shapeAsString(expShape);
+
+    REQUIRE_TRUE(ShapeUtils::shapeAsString(mean)     == expShapeStr, 0, "BATCHNORM_BP op: wrong shape of mean array, expected is %s, but got %s instead !", expShapeStr.c_str(), ShapeUtils::shapeAsString(mean).c_str());
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp]: Stop doing this please.

Comment (raver119) [reply libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp]: String creation just for comparison is unacceptable.

**Review by raver119 — CHANGES_REQUESTED** [review]
Almost LGTM. Few changes required.

**Comment by shyrma** [comment]
corresponding corrections have been made

**Review by raver119 — APPROVED** [review]
